samples/js-generate-synthetic-db/cloud-functions-version/src/index.ts

```diff
@@ -0,0 +1,253 @@
+function getRandomIntInclusive(min: number, max: number): number {
+  // Ensure the minimum is less than or equal to the maximum
+  min = Math.ceil(min);
+  max = Math.floor(max);
+
+  // Generate a random number within the range
+  return Math.floor(Math.random() * (max - min + 1)) + min; 
+}
+  
+  
+// Define dogfood menu item details to be used as input for the data generation.
+const menuItems = [
+  "Doggo chicken instant ramen is sold for $7.99.",
+  "Doggo cheese burger is sold for $8.99.",
+  "Doggo shrimp fried rice is sold for $8.99.",
+  "Doggo pulled pork tacos is sold for $7.99.",
+  "Doggo NY style pizza is sold for $6.99.",
+  "Doggo Arepa is sold for $6.99."
+]
+
+
+// Define a Zod schema to validate the structure of Bone Appetit sales data.
+const BoneAppetitSalesDatabaseSchema = z.object({
+  item: z.string().describe('Name of the dog food item the customer has purchased.'),
+  quantity: z.number().int().describe('Number of items the customer has purchased.'),
+  total_sales: z.number().describe('Total cost of the items in the order.'),
+  status: z.string().describe('Status of the order such as "Pending," "Processing," "Shipped," "Delivered," "Cancelled"'),
+  created_at: z.string().describe('Date and time when the order was placed, NULL if not yet created. The date and time must be after the year 2022.'),
+  shipped_at: z.string().describe('Date and time when the order was shipped, NULL if not yet shipped. The date and time must be after the year 2022.'),
+  delivered_at: z.string().describe('Date and time when the order was delivered, NULL if not yet delivered. The date and time must be after the year 2022.'),
+  gender: z.string().describe("Customer's gender. It can be either Female or Male."),
+  customer_rating: z.number().describe('Rating given by the customer. Rating range is between 1 to 5.'),
+  customer_review: z.string().describe('Authentic, insightful, fun, honest and unique review from a valued customer based on the given rating.'),
+});
+
+// Define a GenKit flow to create Bone Appetit sales data rows using the Gemini 1.5 Pro model.
+const createBoneAppetitSalesRowSchema = defineFlow({
+  name: "createBoneAppetitSalesRowSchema",
+  inputSchema: z.string(),
+  outputSchema: BoneAppetitSalesDatabaseSchema, // Ensure this schema is well-defined
+}, async (input) => {
+  const result = await generate({
+    model: gemini15ProPreview,
+    config: { temperature: 0.3, maxOutputTokens: 8192},
+    prompt: `Generate one unique row of a dataset table at a time. Dataset description: This is a dog food sales database with reviews. Here is the item and its price: ${input}. The customer rating is: ${getRandomIntInclusive(0,5)}. Please ensure the customer review matches the given rating. If the rating is less than 3, please give an honest bad review.`,
+    output: {format:'json', schema: BoneAppetitSalesDatabaseSchema }
+  });
+
+  // 1. Get the parsed result
+  const BoneAppetitSalesDatabase = result.output();
```

Comment:
This should be `BoneAppetitSaleRowItem` rather than `BaloneAppetitSalesDatabase`